code/datums/components/item_two_handed.dm

```diff
@@ -0,0 +1,282 @@
+/**
+ * Two Handed Component
+ *
+ * When applied to an item it will make it two handed
+ *
+ */
+/datum/component/two_handed
+	var/wielded = FALSE 			/// Are we holding the two handed item properly
+	var/force_wielded = 0	 		/// The force of the item when weilded
+	var/force_unwielded = 0		 	/// The force of the item when unweilded
+	var/wieldsound = FALSE 			/// Play sound when wielded
+	var/unwieldsound = FALSE 		/// Play sound when unwielded
+	var/require_twohands = FALSE	/// Does it have to be held in both hands
+
+/**
+ * Two Handed component
+ *
+ * vars:
+ * * require_twohands (optional) Does the item need both hands to be carried
+ * * wieldsound (optional) The sound to play when wielded
+ * * unwieldsound (optional) The sound to play when unwielded
+ * * force_wielded (optional) The force setting when the item is wielded
+ * * force_unwielded (optional) The force setting when the item is unwielded
+ */
+/datum/component/two_handed/Initialize(require_twohands=FALSE, wieldsound=FALSE, unwieldsound=FALSE, force_wielded=0, force_unwielded=0)
+	src.require_twohands = require_twohands
+	src.wieldsound = wieldsound
+	src.unwieldsound = unwieldsound
+	src.force_wielded = force_wielded
+	src.force_unwielded = force_unwielded
+
+	if(isitem(parent))
```

Comment:
If it's not an item `return COMPONENT_INCOMPATIBLE`

Reply:
This is fixed